setup.py

```diff
@@ -44,9 +48,12 @@ def finalize_options(self):
         author="Joshua Arnott",
         author_email="josh@snorfalorpagus.net",
         url="https://github.com/pywr/pywr",
-        setup_requires=["setuptools>=18.0", "setuptools_scm", "cython", "numpy",],
+        setup_requires=["setuptools>=18.0", "setuptools_scm", "cython", "numpy"],
         install_requires=["pandas", "networkx", "scipy", "tables", "xlrd", "packaging", "matplotlib", "jinja2", "ipython"],
-        extras_require={"test": ["pytest"]},
```

Comment:
setuptools docs says this:

> if a project ends up not needing any other packages to support a feature, it should keep an empty requirements list for that feature in its extras_require argument, so that packages depending on that feature don’t break (due to an invalid feature name

But maybe not necessary here? I'm not aware of any other packages that depend on pywr

Reply:
Because I've renamed it? I was in two minds about this. We could still call it "test" if you prefer.

Reply:
We should probably have one like this too: `"optimisation": ["platypus-opt", "pygmo"]`

Reply:
On the keys, I think its just best practice to retain keys so that it doesn't break any downstream packages. Probably also clearer as `test`

I agree with having an optimisation entry.

Reply:
I've kept "dev", put "test" back and added "optimisation".